release-common-lib/src/main/java/org/reactome/util/compare/DBObjectComparer.java

```diff
@@ -147,225 +176,428 @@ public static int compareInstances(
 	 * that will count as 1 diff and the elements will NOT be compared.
 	 */
 	private static int compareInstances(
-		GKInstance inst1, GKInstance inst2, StringBuilder sb, int diffCount, int recursionDepth, int maxRecursionDepth,
-		Predicate<? super SchemaAttribute> customAttributeNameFilter, boolean checkReferrers
+		GKInstance instance1, GKInstance instance2, StringBuilder stringBuilder, int diffCount, int recursionDepth,
+		int maxRecursionDepth, Predicate<? super SchemaAttribute> customAttributeNameFilter, boolean checkReferrers
 	)
 	{
-		int count = diffCount;
+		if (instance1 == null || instance2 == null || isInstanceEdit(instance1) || isInstanceEdit(instance2))
+		{
+			return diffCount;
+		}
 
-		if (inst1 != null && inst2 != null)
+		if (differentInstanceTypes(instance1, instance2))
 		{
-			SchemaClass class1 = inst1.getSchemClass();
-			SchemaClass class2 = inst2.getSchemClass();
+			stringBuilder
+			.append(getIndentString(recursionDepth))
+			.append(getInstanceTypeMismatchMessage(instance1, instance2));
 
-			if (!class1.getName().equals("InstanceEdit") && !class2.getName().equals("InstanceEdit"))
-			{
-				String[] indentArray = new String[recursionDepth * 2];
-				Arrays.fill(indentArray, " ");
-				String indentString = String.join("", indentArray);
+			return 1;
+		}
 
-				if (!class1.getName().equals(class2.getName()))
```

Comment:
Is the output the same if this condition is removed?

Reply:
The condition was actually just relocated to line 188: https://github.com/reactome/data-release-pipeline/pull/121/files/600bdc5e31fbe03661507750fb03f8b195feabe1#diff-b4b21932944d31bdd5550b507e7e2195R188